### ISSUE TITLE
Random Check Failure Behind HTTP Proxy

### ISSUE THREAD

**Issue by zicklag** [EmbarkStudios/cargo-deny#305]
**Describe the bug**

When running `cargo deny check` on a computer behind an HTTP proxy that requires authentication I get this error about 50% of the time:

```
cargo-deny: libgit2/src/transports/httpclient.c:926: proxy_connect: Assertion `client->state == DONE' failed.
fish: “cargo deny check” terminated by signal SIGABRT (Abort)
```

**To Reproduce**
1. Set `http_proxy` and `https_proxy` environment variables in the format `http://user:pass@server:port` poin
2. Run `cargo deny check` in a project

**Expected behavior**
No error should be displayed and the process should not abort.

**Device:**
 - OS: Ubuntu 20.04 ( Pop!_OS variant )

**Additional context**
I've used cargo deny a while ago ( ~ 6 months, maybe :man_shrugging: ) and I didn't have that problem, then.

**Comment by Jake-Shadle** [comment]
We updated to the latest libgit2 (1.1.0) in the last release, a brief look indicates there is an [existing issue[(https://github.com/libgit2/libgit2/issues/5452) from March (there have been several releases made between though) that seems to be the same issue you are seeing.

**Comment by zicklag** [comment]
Cool, thanks for looking into it. I'll leave a comment on that thread. :+1:

**Comment by Jake-Shadle** [comment]
This was _possibly_ fixed in #520
